### DISCUSSION TITLE
.NET 7.0 Support

### DISCUSSION THREAD

**Discussion by BroMarduk** [betalgo/openai#222] [Ideas]
I took a first pass at getting the code to compile without warnings in .NET 7 (mostly additional nullable support) and have something working well for me (.NET 6 and .NETSTANDARD 2.0 also working).  I've only tested with the playground but would be happy to upload in a new branch if there is interest and someone wants to validate some nullable support choices.

<img width="675" alt="Screenshot 2023-04-12 005006" src="https://user-images.githubusercontent.com/2371774/231353014-d85b9482-e616-49c9-8886-5c37218661c9.png">

<img width="716" alt="Screenshot 2023-04-12 005140" src="https://user-images.githubusercontent.com/2371774/231353206-fc90e9ef-cf38-4a6c-ad24-76eb45934f39.png">

**Comment by Alerinos** [comment]
I'm using in .net 8 and everything works fine. If you have any problem and errors, it's only with you.

**Comment by BroMarduk** [comment]
It does "work" from a practical sense, but then I have to deploy dependencies from two different .NET frameworks.  Having a native version of the NuGet package for .NET 7 or even the .NET 8 Preview is almost always going to be the better option.

**Comment by kayhantolga** [comment]
Hey, @BroMarduk it would be nice to have .NET 7 and .NET 8 Preview support. If you can create a PR we can add this feature. 
And what do you mean by "validate some nullable support choices"?

**Reply by BroMarduk** [reply]
I made the PR for .NET 7.  There are multiple commits, since I wanted to fix some of the complier warnings as well, most of those around whether to make a field nullable or use a default value. (nullable support choices I mentioned.)

We are using the code in a production environment now, so I had a need to remove the warnings.

**Reply by kayhantolga** [reply]
I will review your pull request. Thank you for contributing! I've been meaning to clean up the warnings, but I've been a bit lazy about it. I'm glad to hear that someone has already taken care of it 😅 
Please also share your project so that I can add more projects to the wiki page
